azure/scope/managedmachinepool.go

```diff
@@ -224,6 +224,46 @@ func buildAgentPoolSpec(managedControlPlane *infrav1.AzureManagedControlPlane,
 		}
 	}
 
+	if managedMachinePool.Spec.LinuxOSConfig != nil {
```

Comment:
do you still need to do this? or could this just become `agentPoolSpec.LinuxOSConfig = managedMachinePool.Spec.LinuxOSConfig` since you're using the same type now?

Reply:
Oh yes, thank you for pointing it out. Refactoring it.